cmd/kubeadm/app/util/kubeconfig/kubeconfig.go

```diff
@@ -68,6 +69,10 @@ func CreateWithToken(serverURL, clusterName, userName string, caCert []byte, tok
 
 // ClientSetFromFile returns a ready-to-use client from a KubeConfig file
 func ClientSetFromFile(path string) (*clientset.Clientset, error) {
+	if len(path) == 0 {
+		// get the default kubeconfig from env
+		path = os.Getenv("KUBECONFIG")
```

Comment:
if `KUBECONFIG` is `""`, and `clientcmd.LoadFromFile` will return the error.  People may use `--kubeconfig=invalidpath` too.